---
---
title: "Woolly_DispersalPipeline"
author: "Sofia M. Weaver"
date: "`r Sys.Date()`"
output: html_document
---
```{r,echo=F}
library(knitr)
```

```{css,echo=F}
/* Style for Bash code blocks */
.terminalCode {
    background-color: black !important;
    color: white !important;
}
```

## Background 

Ok, this doc is just so I can have some real data for my honors thesis defense (eek) investigating if there is sex-biased dispersal in the Critically Endangered Yellow-Tailed Woolly Monkey (L. flavicauda). So we'll worry about HQ-rebasecalling later. For now, let's just work with rapid basecall data that passed. Thankfully, since Chris is the best advisor ever (hi Chris), I have an SRY starter pipeline that he wrote for me while we were at Los Amigos together. I'm going to be using that moving forward, so a lot of this code will look very familiar, with updated paths. 

```{bash define-black-chunks, eval=F, echo=T, class.source="sccCode"}
Black code chunks like this are either UNIX or python code, and are meant to be executed in your computer Terminal.
```

```{r define-grey-chunks, echo=T,eval=F}
Grey code chunks like this are meant to be executed in *R/Rstudio*
```

## Concatenating the sequences

Alright, let's get started and concatenate: 

```{bash concatenate, eval=F, echo=T, class.source="sccCode"}
    # for each directory in our specified condition (remember that * is a wildcard):
for dir in /Users/sofia/Documents/WoollyRun9/*/;
do
    # ${variable%something} is the part of $variable
    # before the string "something"
    # basename path/to/file is the name of the file
    # without the full path
    # $(some command) allows you to use command result as a string
    # Combining the above, we form a string based on our fasta file
    # This string can be useful to name stuff in a clean manner later
  barcode_name=$(basename ${dir%})
  echo ${barcode_name}
    # Create a directory for the results for this sequence
    # -p option avoids a failure in case the directory already exists
  mkdir -p  ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/${barcode_name}
  cd ${dir}
  cat $(ls -t) > ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/${barcode_name}/WoollyRun9_${barcode_name}.fastq.gz;
done
```


Next, we’ll get the stats (this might take a while, so be patient):

```{bash stats, eval=F, echo=T, class.source="terminalCode"}
for runfile in ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/*/*.fastq.gz;
do
    # Run seqkit stats on each file and append to a single document
    seqkit stats ${runfile} >> ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/Reads_WoollyRun9_QC1.txt
done

# Remove duplicate headers and save as new file
sed '2,${/file/d;}' ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/Reads_WoollyRun9_QC1.txt > ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/Reads_WoollyRun9_QC.txt

# Delete the temporary file
rm ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/Reads_WoollyRun9_QC1.txt
```

## Indexing our reference

Let’s make a minimap2 index of the Alouatta pigra SRY sequence that we’ll use as our reference:

**NOTE** I just used the fasta file Chris made when he originally was doing this, but included this chunk of code for overall thought process. 

```{bash indexing reference, eval=F, echo=T, class.source="terminalCode"}
minimap2 -d ~/Documents/SMAGL/woolly_DispersalPipeline/Alouatta_pigra.SRY.CortezOrtiz.mmi ~/Documents/SMAGL/woolly_DispersalPipeline/Alouatta_pigra.SRY.CortezOrtiz.fasta
```

## Mapping to the reference

Here we map all reads to Alouatta pigra SRY, outputting them into a single sam file:

```{bash mapping reference BC12, eval=F, echo=T, class.source="terminalCode"}
minimap2 -ax map-ont ~/Documents/SMAGL/woolly_DispersalPipeline/Alouatta_pigra.SRY.CortezOrtiz.fasta ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/barcode12/WoollyRun9_barcode12.fastq.gz -a -o ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/barcode12/SRY_barcode12_QCreads_vs_ApigSRY.sam -t 24
```

And here we sort and filter for mapped reads only and convert them to bam files:

```{bash bam conversion, eval=F, echo=T, class.source="terminalCode"}
samtools sort ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/barcode12/SRY_barcode12_QCreads_vs_ApigSRY.sam -o ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/barcode12/SRY_barcode12_QCreads_vs_ApigSRY.bam 

samtools view -F 0x4 -b ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/barcode12/SRY_barcode12_QCreads_vs_ApigSRY.bam  > ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/barcode12/SRY_barcode12_QCreads_vs_ApigSRY.aln.bam 

samtools index ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/barcode12/SRY_barcode12_QCreads_vs_ApigSRY.aln.bam
```


Convert our alignment to FASTQ:

```{bash FASTQ conversion, eval=F, echo=T, class.source="terminalCode"}
samtools bam2fq ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/barcode12/SRY_barcode12_QCreads_vs_ApigSRY.aln.bam > ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/barcode12/SRY_barcode12_QCreads_vs_ApigSRY.aln.fastq
```

```{bash, eval = F, echo = T, class.source="terminalCode"}
seqkit stats ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/barcode12/SRY_barcode12_QCreads_vs_ApigSRY.aln.fastq

# FASTQ   DNA      8,124  3,497,105      157    430.5      875
```

## Coverage

Ok, so using the Alouatta pigra sequence gets us a few more reads… that’s good! Let’s take a look at coverage:

```{bash coverage, eval=F, echo=T, class.source="terminalCode"}
samtools depth ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/barcode12/SRY_barcode12_QCreads_vs_ApigSRY.aln.bam > ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/barcode12/SRY_barcode12_QCreads_vs_ApigSRY.coverage
```

Then, we import into R and do a few manipulations to graph this output so we can visualize depth of coverage across the Alouatta pigra SRY amplicon by position:

```{r coverage BC12}
coverageBC12Apig=read.table("~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/barcode12/SRY_barcode12_QCreads_vs_ApigSRY.coverage", sep="\t", header=F)
```


```{r coverage fig}
library(reshape)
library(tidyverse)

coverageBC12Apig <-
  coverageBC12Apig %>%
  rename("Chr" = 1, "locus" = 2, "depth" = 3)

ggplot(coverageBC12Apig, aes(x=locus, y=depth)) + theme_bw() +
geom_point(colour="darkgreen", size=1, shape=20, alpha=1/3) +
scale_y_continuous(trans = scales::log10_trans(), breaks = scales::trans_breaks("log10", function(x) 10^x))
```

## Consensus

Now we need to create a consensus sequences for each barcoded individual. To do this, we'll use a program called [{NGSpeciesID}](https://github.com/ksahlin/NGSpeciesID) (created by Stefan Prost; you may have met him at Los Amigos last summer!).

#### Creating the NGSpeciesID Environment

**NOTE: On the SMAGL Laptop this is already installed!**
***I RECOMMEND RUNNING ALL CHUNKS THAT USE NGSpeciesID ON THE SMAGL LAPTOP IF YOU ARE UNSABLE TO DOWNLOAD*** 

To get started, you'll need to install {NGSpeciesID} into a `conda` environment using Terminal

```{bash, eval = F, echo = T, class.source="terminalCode"}
conda create --yes -n NGSpeciesID python==3.10 pip #create virtual environment
conda init
conda activate NGSpeciesID #activate the virtual environment
pip install --upgrade pip #update pip

conda install --yes -c conda-forge -c bioconda -c nanoporetech openblas spoa minimap2 nomkl #install the rest

/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
(echo; echo 'eval "$(/opt/homebrew/bin/brew shellenv)"') >> /Users/"${USER}"/.zprofile
eval "$(/opt/homebrew/bin/brew shellenv)" #install home brew in virtual env

brew install openssl #install openssl
export LIBRARY_PATH=/opt/homebrew/lib
pip install medaka==1.11.3 #install medaka

pip install NGSpeciesID

pip install tensorflow-macos==2.10.0

conda deactivate #takes you back to your (base) computer environment
```


NOTE: You may also want to install `libmamba` if your `conda` environment gets stuck "solving the environment"... it will then tell you where incompatibilities exist between package versions so you can troubleshoot:

```{bash, eval = F, echo = T, class.source="terminalCode"}
conda install -n base conda-libmamba-solver
```


let’s run these to get a consensus sequence for BC12:
**NOTE:** I RAN THIS ON THE SMAGL LAPTOP. THIS WILL NOT WORK IF YOU DO NOT HAVE NGSpeciesID INSTALLED

```{bash consensus BC12, eval=F, echo=T, class.source="terminalCode"}
conda activate NGSpeciesID

NGSpeciesID --ont --m 300 --s 150 --consensus --medaka --primer_file  /Users/smaglschmittlab/Documents/SMAGL/Woolly_DispersalPipeline/SRYprimers.txt \
--fastq /Users/smaglschmittlab/Documents/SMAGL/Woolly_DispersalPipeline/sequencing_data/WoollyRun9/barcode12/SRY_barcode12_QCreads_vs_ApigSRY.aln.fastq \
--outfolder /Users/smaglschmittlab/Documents/SMAGL/Woolly_DispersalPipeline/sequencing_data/WoollyRun9/barcode12/barcode12QC_Apig_300e150_consensus

conda deactivate
```


# All Barcodes
Let's map all other barcodes individually for now since there are only 5.. but really need to figure out that loop for all of them. 

```{bash mapping reference all BC, eval=F, echo=T, class.source="terminalCode"}
minimap2 -ax map-ont ~/Documents/SMAGL/woolly_DispersalPipeline/Alouatta_pigra.SRY.CortezOrtiz.fasta ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/barcode13/WoollyRun9_barcode13.fastq.gz -a -o ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/barcode13/SRY_barcode13_QCreads_vs_ApigSRY.sam -t 24

minimap2 -ax map-ont ~/Documents/SMAGL/woolly_DispersalPipeline/Alouatta_pigra.SRY.CortezOrtiz.fasta ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/barcode14/WoollyRun9_barcode14.fastq.gz -a -o ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/barcode14/SRY_barcode14_QCreads_vs_ApigSRY.sam -t 24

minimap2 -ax map-ont ~/Documents/SMAGL/woolly_DispersalPipeline/Alouatta_pigra.SRY.CortezOrtiz.fasta ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/barcode15/WoollyRun9_barcode15.fastq.gz -a -o ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/barcode15/SRY_barcode15_QCreads_vs_ApigSRY.sam -t 24

minimap2 -ax map-ont ~/Documents/SMAGL/woolly_DispersalPipeline/Alouatta_pigra.SRY.CortezOrtiz.fasta ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/barcode16/WoollyRun9_barcode16.fastq.gz -a -o ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/barcode16/SRY_barcode16_QCreads_vs_ApigSRY.sam -t 24
```

Sort and filter for mapped reads only and convert them to bam files:

```{bash all BC bam conversion, eval=F, echo=T, class.source="terminalCode"}
samtools sort ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/barcode13/SRY_barcode13_QCreads_vs_ApigSRY.sam -o ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/barcode13/SRY_barcode13_QCreads_vs_ApigSRY.bam 

samtools view -F 0x4 -b ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/barcode13/SRY_barcode13_QCreads_vs_ApigSRY.bam  > ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/barcode13/SRY_barcode13_QCreads_vs_ApigSRY.aln.bam 

samtools index ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/barcode13/SRY_barcode13_QCreads_vs_ApigSRY.aln.bam



samtools sort ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/barcode14/SRY_barcode14_QCreads_vs_ApigSRY.sam -o ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/barcode14/SRY_barcode14_QCreads_vs_ApigSRY.bam 

samtools view -F 0x4 -b ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/barcode14/SRY_barcode14_QCreads_vs_ApigSRY.bam  > ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/barcode14/SRY_barcode14_QCreads_vs_ApigSRY.aln.bam 

samtools index ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/barcode14/SRY_barcode14_QCreads_vs_ApigSRY.aln.bam



samtools sort ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/barcode15/SRY_barcode15_QCreads_vs_ApigSRY.sam -o ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/barcode15/SRY_barcode15_QCreads_vs_ApigSRY.bam 

samtools view -F 0x4 -b ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/barcode15/SRY_barcode15_QCreads_vs_ApigSRY.bam  > ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/barcode15/SRY_barcode15_QCreads_vs_ApigSRY.aln.bam 

samtools index ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/barcode15/SRY_barcode15_QCreads_vs_ApigSRY.aln.bam



samtools sort ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/barcode16/SRY_barcode16_QCreads_vs_ApigSRY.sam -o ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/barcode16/SRY_barcode16_QCreads_vs_ApigSRY.bam 

samtools view -F 0x4 -b ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/barcode16/SRY_barcode16_QCreads_vs_ApigSRY.bam  > ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/barcode16/SRY_barcode16_QCreads_vs_ApigSRY.aln.bam 

samtools index ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/barcode16/SRY_barcode16_QCreads_vs_ApigSRY.aln.bam
```


Convert alignments to FASTQ:

```{bash all BC FASTQ conversion, eval=F, echo=T, class.source="terminalCode"}
samtools bam2fq ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/barcode13/SRY_barcode13_QCreads_vs_ApigSRY.aln.bam > ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/barcode13/SRY_barcode13_QCreads_vs_ApigSRY.aln.fastq


samtools bam2fq ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/barcode14/SRY_barcode14_QCreads_vs_ApigSRY.aln.bam > ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/barcode14/SRY_barcode14_QCreads_vs_ApigSRY.aln.fastq


samtools bam2fq ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/barcode15/SRY_barcode15_QCreads_vs_ApigSRY.aln.bam > ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/barcode15/SRY_barcode15_QCreads_vs_ApigSRY.aln.fastq


samtools bam2fq ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/barcode16/SRY_barcode16_QCreads_vs_ApigSRY.aln.bam > ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/barcode16/SRY_barcode16_QCreads_vs_ApigSRY.aln.fastq
```


```{bash, eval = F, echo = T, class.source="terminalCode"}
seqkit stats ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/barcode13/SRY_barcode13_QCreads_vs_ApigSRY.aln.fastq

seqkit stats ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/barcode14/SRY_barcode14_QCreads_vs_ApigSRY.aln.fastq

seqkit stats ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/barcode15/SRY_barcode15_QCreads_vs_ApigSRY.aln.fastq

seqkit stats ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/barcode16/SRY_barcode16_QCreads_vs_ApigSRY.aln.fastq

# 13: FASTQ   DNA     28,582  12,329,169      174    431.4    1,095
# 14: FASTQ   DNA         33   13,962      324    423.1      453
# 15: FASTQ   DNA      2,080  901,294      197    433.3      764
# 16: FASTQ   DNA        221   94,583      248      428      646
```


## Coverage for all BCs

 Let’s take a look at coverage for the rest of our barcdoes:

```{bash coverage all BCs, eval=F, echo=T, class.source="terminalCode"}
samtools depth ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/barcode13/SRY_barcode13_QCreads_vs_ApigSRY.aln.bam > ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/barcode13/SRY_barcode13_QCreads_vs_ApigSRY.coverage


samtools depth ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/barcode14/SRY_barcode14_QCreads_vs_ApigSRY.aln.bam > ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/barcode14/SRY_barcode14_QCreads_vs_ApigSRY.coverage


samtools depth ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/barcode15/SRY_barcode15_QCreads_vs_ApigSRY.aln.bam > ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/barcode15/SRY_barcode15_QCreads_vs_ApigSRY.coverage


samtools depth ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/barcode16/SRY_barcode16_QCreads_vs_ApigSRY.aln.bam > ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/barcode16/SRY_barcode16_QCreads_vs_ApigSRY.coverage
```


Then, we import into R and do a few manipulations to graph this output so we can visualize depth of coverage across the Alouatta pigra SRY amplicon by position:

```{r coverage all BC}
coverageBC13Apig=read.table("~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/barcode13/SRY_barcode13_QCreads_vs_ApigSRY.coverage", sep="\t", header=F)

coverageBC14Apig=read.table("~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/barcode14/SRY_barcode14_QCreads_vs_ApigSRY.coverage", sep="\t", header=F)

coverageBC15Apig=read.table("~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/barcode15/SRY_barcode15_QCreads_vs_ApigSRY.coverage", sep="\t", header=F)

coverageBC16Apig=read.table("~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/barcode16/SRY_barcode16_QCreads_vs_ApigSRY.coverage", sep="\t", header=F)
```

I know I already plotted it, but I'm including BC12 again so so we can see coverage for all samples at once. 

```{r coverage fig all BCs}
library(reshape)
library(tidyverse)


coverageBC12Apig <- coverageBC12Apig %>%
  rename_with(~ c("Chr", "locus", "depth"), .cols = 1:3)

ggplot(coverageBC12Apig, aes(x=locus, y=depth)) + theme_bw() +
geom_point(colour="darkgreen", size=1, shape=20, alpha=1/3) +
scale_y_continuous(trans = scales::log10_trans(), breaks = scales::trans_breaks("log10", function(x) 10^x)) +
  ggtitle("BC12")



coverageBC13Apig <- coverageBC13Apig %>%
  rename_with(~ c("Chr", "locus", "depth"), .cols = 1:3)

ggplot(coverageBC13Apig, aes(x=locus, y=depth)) + theme_bw() +
geom_point(colour="darkgreen", size=1, shape=20, alpha=1/3) +
scale_y_continuous(trans = scales::log10_trans(), breaks = scales::trans_breaks("log10", function(x) 10^x)) +
  ggtitle("BC13")


coverageBC14Apig <- coverageBC14Apig %>%
  rename_with(~ c("Chr", "locus", "depth"), .cols = 1:3)

ggplot(coverageBC14Apig, aes(x=locus, y=depth)) + theme_bw() +
geom_point(colour="darkgreen", size=1, shape=20, alpha=1/3) +
scale_y_continuous(trans = scales::log10_trans(), breaks = scales::trans_breaks("log10", function(x) 10^x)) +
  ggtitle("BC14")


coverageBC15Apig <- coverageBC15Apig %>%
  rename_with(~ c("Chr", "locus", "depth"), .cols = 1:3)

ggplot(coverageBC15Apig, aes(x=locus, y=depth)) + theme_bw() +
geom_point(colour="darkgreen", size=1, shape=20, alpha=1/3) +
scale_y_continuous(trans = scales::log10_trans(), breaks = scales::trans_breaks("log10", function(x) 10^x)) +
  ggtitle("BC15")


coverageBC16Apig <- coverageBC16Apig %>%
  rename_with(~ c("Chr", "locus", "depth"), .cols = 1:3)

ggplot(coverageBC16Apig, aes(x=locus, y=depth)) + theme_bw() +
geom_point(colour="darkgreen", size=1, shape=20, alpha=1/3) +
scale_y_continuous(trans = scales::log10_trans(), breaks = scales::trans_breaks("log10", function(x) 10^x)) +
  ggtitle("BC16")
```






## Looping all BCs 
forgot to check with this with Chris and coudn't get it to work. again. it's crunch time so i'm doing each individually since there are only 5. I DO NOT RECOMMEND doing this for a larger sample size. This would be impossible with the entire sequencing data. 

```{bash mapping all BCs, eval=F, echo=T, class.source="terminalCode"}
for dir in /Users/sofia/Documents/WoollyRun9*;
do
    # First let's pull out the barcode name:
    barcode_name=$(basename ${dir%}) #here we get it
    echo ${barcode_name} #and we'll print the name as we go along

    # Now we'll map to the Alouatta pigra reference:
    minimap2 -ax map-ont    ~/Documents/SMAGL/woolly_DispersalPipeline/Alouatta_pigra.SRY.CortezOrtiz.fasta${dir}/WoollyRun9_${barcode_name}.fastq.gz -a -o ${dir}/SRY_${barcode_name}_QCreads_vs_ApigSRY.sam -t 24

    # Sort and filter for mapped reads:
    samtools sort ${dir}/SRY_${barcode_name}_QCreads_vs_ApigSRY.sam -o ${dir}/SRY_${barcode_name}_QCreads_vs_ApigSRY.bam 
    samtools view -F 0x4 -b ${dir}/SRY_${barcode_name}_QCreads_vs_ApigSRY.bam  > ${dir}/SRY_${barcode_name}_QCreads_vs_ApigSRY.aln.bam 
    samtools index ${dir}/SRY_${barcode_name}_QCreads_vs_ApigSRY.aln.bam

    # Convert alignment to FASTQ:
    samtools bam2fq ${dir}/SRY_${barcode_name}_QCreads_vs_ApigSRY.aln.bam > ${dir}/SRY_${barcode_name}_QCreads_vs_ApigSRY.aln.fastq

    # Generate coverage file:
    samtools depth ${dir}/SRY_${barcode_name}_QCreads_vs_ApigSRY.aln.bam > ${dir}/SRY_${barcode_name}_QCreads_vs_ApigSRY.coverage;

done
```

# making consensus sequences for all Barcodes

**NOTE:** I RAN THIS CODE ON THE SMAGL LAPTOP. THIS CODE WILL NOT WORK IF YOU DO NOT HAVE NGSpeciesID installed

```{bash all BC consensus, eval=F, echo=T, class.source="terminalCode}
conda activate NGSpeciesID

NGSpeciesID --ont --m 300 --s 150 --consensus --medaka --primer_file  /Users/smaglschmittlab/Documents/SMAGL/Woolly_DispersalPipeline/SRYprimers.txt \
--fastq /Users/smaglschmittlab/Documents/SMAGL/Woolly_DispersalPipeline/sequencing_data/WoollyRun9/barcode13/SRY_barcode13_QCreads_vs_ApigSRY.aln.fastq \
--outfolder /Users/smaglschmittlab/Documents/SMAGL/Woolly_DispersalPipeline/sequencing_data/WoollyRun9/barcode13/barcode13QC_Apig_300e150_consensus


NGSpeciesID --ont --m 300 --s 150 --consensus --medaka --primer_file  /Users/smaglschmittlab/Documents/SMAGL/Woolly_DispersalPipeline/SRYprimers.txt \
--fastq /Users/smaglschmittlab/Documents/SMAGL/Woolly_DispersalPipeline/sequencing_data/WoollyRun9/barcode14/SRY_barcode14_QCreads_vs_ApigSRY.aln.fastq \
--outfolder /Users/smaglschmittlab/Documents/SMAGL/Woolly_DispersalPipeline/sequencing_data/WoollyRun9/barcode14/barcode14QC_Apig_300e150_consensus


NGSpeciesID --ont --m 300 --s 150 --consensus --medaka --primer_file  /Users/smaglschmittlab/Documents/SMAGL/Woolly_DispersalPipeline/SRYprimers.txt \
--fastq /Users/smaglschmittlab/Documents/SMAGL/Woolly_DispersalPipeline/sequencing_data/WoollyRun9/barcode15/SRY_barcode15_QCreads_vs_ApigSRY.aln.fastq \
--outfolder /Users/smaglschmittlab/Documents/SMAGL/Woolly_DispersalPipeline/sequencing_data/WoollyRun9/barcode15/barcode15QC_Apig_300e150_consensus


NGSpeciesID --ont --m 300 --s 150 --consensus --medaka --primer_file  /Users/smaglschmittlab/Documents/SMAGL/Woolly_DispersalPipeline/SRYprimers.txt \
--fastq /Users/smaglschmittlab/Documents/SMAGL/Woolly_DispersalPipeline/sequencing_data/WoollyRun9/barcode16/SRY_barcode16_QCreads_vs_ApigSRY.aln.fastq \
--outfolder /Users/smaglschmittlab/Documents/SMAGL/Woolly_DispersalPipeline/sequencing_data/WoollyRun9/barcode16/barcode16QC_Apig_300e150_consensus

conda deactivate
```


## Closing Out `conda`

Once that's done, we need to go back to our base environment:
```{bash, eval = F, echo = T, class.source="sccCode"}
conda deactivate
```

You can also erase your environment (if you think you'll never use it again) by entering:
```{bash, eval = F, echo = T, class.source="sccCode"}
conda remove --name NGSpeciesID --all
```



## BC Alignment

### Getting All BC Together

Let's try aligning our sequences from this run, to see if there's consistency between them:
```{r}
#This loop creates a list of all the medaka-polished consensus sequences in our barcode folders:
barcodes<-c("barcode12","barcode13","barcode14","barcode15","barcode16")

library(Biostrings)
multibarcode.300_150<-c()
for (barcode in barcodes) {
  a<-paste0("~/Documents/SMAGL/Woolly_DispersalPipeline/sequencing_data/WoollyRun9/",barcode,
            "/",barcode,"QC_Apig_300e150_consensus/medaka_cl_id_*/consensus.fasta")
  
  assign(paste0(barcode,"_150con"),Biostrings::readDNAStringSet(Sys.glob(a)))
  
  multibarcode.300_150<-c(multibarcode.300_150,get(paste0(barcode,"_150con")))  
}

#And this concatenates (unlists) all the FASTA files into one file:
names(multibarcode.300_150) <- NULL
multibarcode.300_150<-do.call(c, multibarcode.300_150)
names(multibarcode.300_150)<-c("barcode12","barcode13","barcode14","barcode15","barcode16")
```

We can also see which are identical to each other:
```{r}
duplicated(multibarcode.300_150)
```


### Aligning All Seqs

```{r}
#And now we can make an alignment:
aln.barcode300_150<-rMSA::mafft(multibarcode.300_150)
```


### Phylogeny of All Seqs

Let's make a tree:
```{r}
library(phangorn)
barcode300_150.phyDat<-as.phyDat(aln.barcode300_150)
dm  <- dist.ml(barcode300_150.phyDat)
treeUPGMA  <- upgma(dm)
treeNJ  <- NJ(dm)
fun <- function(x) { upgma(dist.ml(x)) }
bs_upgma <- bootstrap.phyDat(barcode300_150.phyDat, fun)
plotBS(treeUPGMA, bs_upgma, main="UPGMA of SRY 300-150")
```


### Prettifying our Alignment

We can take a closer look at the alignment here. We'll use the [{ggmsa}](http://yulab-smu.top/ggmsa/) package:
```{r,warning=F}
library(ggmsa)

#png(filename="~/Documents/SMAGL/Woolly_DispersalPipeline/sequencing_data/WoollyRun9/SRY_con300_150_aln.trim.png",width = 25, height = 5, units = "in",res=600)

ggmsa::ggmsa(aln.barcode300_150, seq_name = T, color = "Chemistry_NT") + geom_seqlogo(color = "Chemistry_NT") + geom_msaBar()

dev.off()
```


### Reverse Complementing Seqs
alright barcodes 12 and 14 kinda look funky so let's try reversing them. I use the same motif that Chris does in his starter pipeline to automate the detection of likely reverse transcripts.

```{r}
#Let's try automating the detection of likely reverse transcripts. We'll do this by looking for a common DNA motif across all transcripts of one orientation that doesn't contain any indels. Note that you should visually check any new alignment to estimate what the proper motif of the forward strand should be, as there may be indels in your previous motif in new samples.

#This code automatically detects likely reverse complements by noting which sequences are missing the motif 'TTTCCAACTCTGTCAT', which is shared across most sequences. It then isolates them in its multisequence object and reverse complements them all at once:
barcodes150rc<-reverseComplement(multibarcode.300_150[which(vcountPattern('TTTCCAACTCTGTCAT', multibarcode.300_150) < 1)])

names(barcodes150rc)<-paste0(names(barcodes150rc),".rc") #rename seqs with ".rc" at the end

#Let's remove the ones we've reverse complemented from our original object:
multibarcode.300_150.fw<-multibarcode.300_150[which(vcountPattern('TTTCCAACTCTGTCAT', multibarcode.300_150) >= 1)]

#And this concatenates our two objects, the reversed and the original minus reversed:
multibarcode.300_150.rc<-c(multibarcode.300_150.fw,barcodes150rc)

#And here we can reorder the sequences so that same-barcodes align:
multibarcode.300_150.rc<-multibarcode.300_150.rc[order(names(multibarcode.300_150.rc))]
```
Ok let's realign to see how it's looking. 

### Re-Aligning with RC
```{r}
multibarcode.300_150.rc.aln <- rMSA::mafft(multibarcode.300_150.rc)
multibarcode.300_150.rc.aln
```
Nice! this looks better. 

```{r}
multibarcode.300_150.rc.plot <- ggmsa::ggmsa(multibarcode.300_150.rc.aln, seq_name = T, color = "Chemistry_NT") + geom_seqlogo(color = "Chemistry_NT") + geom_msaBar()
multibarcode.300_150.rc.plot

#Below is code to save the alignment image:
png(filename="~/Documents/SMAGL/Woolly_DispersalPipeline/sequencing_data/WoollyRun9/SRY_con300_150_aln.trim.png",width = 30, height = 7, units = "in",res=600)
multibarcode.300_150.rc.plot
dev.off()
```


Alright, I don't think this needs any trimming. Let's keep going. 

### New Phylogeny
```{r}
library(phangorn)

barcode300_150.phyDat<-as.phyDat(multibarcode.300_150.rc.aln)
dm  <- dist.ml(barcode300_150.phyDat)
treeUPGMA  <- upgma(dm)
treeNJ  <- NJ(dm)
fun <- function(x) { upgma(dist.ml(x)) }
bs_upgma <- bootstrap.phyDat(barcode300_150.phyDat, fun)
plotBS(treeUPGMA, bs_upgma, main="UPGMA of SRY 300-150")
```
umm.. ok this doesn't really very informative. Remember this can all change once I use the HQ rebasecalled data! 

## Haplotype Network

```{r}
library(tidyverse)
library(Biostrings)

lagoSRY12<-readDNAStringSet("/Users/sofia/Documents/SMAGL/Woolly_DispersalPipeline/sequencing_data/WoollyRun9/barcode12/barcode12QC_Apig_300e150_consensus/medaka_cl_id_7/consensus.fasta")

lagoSRY13<-readDNAStringSet("/Users/sofia/Documents/SMAGL/Woolly_DispersalPipeline/sequencing_data/WoollyRun9/barcode13/barcode13QC_Apig_300e150_consensus/medaka_cl_id_8/consensus.fasta")

lagoSRY14<-readDNAStringSet("/Users/sofia/Documents/SMAGL/Woolly_DispersalPipeline/sequencing_data/WoollyRun9/barcode14/barcode14QC_Apig_300e150_consensus/medaka_cl_id_0/consensus.fasta")

lagoSRY15<-readDNAStringSet("/Users/sofia/Documents/SMAGL/Woolly_DispersalPipeline/sequencing_data/WoollyRun9/barcode15/barcode15QC_Apig_300e150_consensus/medaka_cl_id_1/consensus.fasta")

lagoSRY16<-readDNAStringSet("/Users/sofia/Documents/SMAGL/Woolly_DispersalPipeline/sequencing_data/WoollyRun9/barcode16/barcode16QC_Apig_300e150_consensus/medaka_cl_id_4/consensus.fasta")

# Combine all consensus sequences into a single DNAStringSet

names(lagoSRY12) <- "barcode12"
names(lagoSRY13) <- "barcode13"
names(lagoSRY14) <- "barcode14"
names(lagoSRY15) <- "barcode15"
names(lagoSRY16) <- "barcode16"


consensus_combined <- c(lagoSRY12, lagoSRY13, lagoSRY14, lagoSRY15, lagoSRY16)


# Write to a new FASTA file
writeXStringSet(consensus_combined, filepath = "/Users/sofia/Documents/SMAGL/Woolly_DispersalPipeline/sequencing_data/SRY_combined_consensus.fasta")
```


```{r}
library(DECIPHER)

lagoSRY<-readDNAStringSet("~/Documents/SMAGL/Woolly_DispersalPipeline/sequencing_data/SRY_combined_consensus.fasta")

lagoSRY <- OrientNucleotides(lagoSRY)

#Run the alignment
lagoSRY.aln.pre <- AlignSeqs(c(lagoSRY))
```

```{r}
length(lagoSRY.aln.pre)
lagoSRY.aln.pre
```

```{r}
SRY.popinfo <-read.csv("~/Documents/SMAGL/woolly_DispersalPipeline/SRY_popinfo.csv", sep = ",", header=TRUE, comment.char = "#")
rownames(SRY.popinfo)<-SRY.popinfo$barcode
SRY.popinfo
```


```{r}
lagoSRY.aln<-lagoSRY.aln.pre[SRY.popinfo$barcode]
length(lagoSRY.aln)
```

Now we’ll load the package we’ll use to construct our haplotype network, called {geneHapR}:

```{r}
library(geneHapR)

SRYhapResult <- seqs2hap(lagoSRY.aln,
                      Ref = names(lagoSRY.aln)[1],
                      hapPrefix = "H",
                      hetero_remove = TRUE,
                      na_drop = TRUE,
                      maxGapsPerSeq = 0.25)

sites(SRYhapResult)
```

8 sites informing the hap results. 

Summary of SRY haplotypes

```{r}
SRYhapSummary <- hap_summary(SRYhapResult)
print(SRYhapSummary)
```
uhhh it's saying they all have their own haplotype? I think this will change once we get the HQ data. 

```{r}
plotHapTable(SRYhapSummary,
             title = "Lagothrix flavicauda SRY Haplotypes")
```

```{r}
SRYhapNet <- get_hapNet(SRYhapSummary,
                     AccINFO = SRY.popinfo,
                     groupName = "collection_site")
```


```{r}
plot(SRYhapNet)


plotHapNet(SRYhapNet,
           size = "freq",                   # circle size
           scale = "log10",                 # scale circle with 'log10(size + 1)'
           cex = 1,                         # size of hap symbol
           col.link = 1,                    # link colors
           link.width = 1,                  # link widths
           show.mutation = 2,               # mutation types one of c(0,1,2,3)
           #hapGroup = 61,                   # draw pie charts for each haplotype
           legend = c(6,6),                # legend position
           show_color_legend = TRUE,
           show_size_legend = TRUE,
           labels = FALSE,
           pie.lim = c(2, 5),
           backGround=c("red3","orange","skyblue"))
```
ok.. don't hate me. but let's do all of this again but with out HQ basecall data (that just finished basecalling)!!



## Again! But with HQ data!!!


## Concatenating the sequences

Alright, let's get started and concatenate. I'm praying that updating pathways isn't going to be a problem. 

```{bash HQ concatenate, eval=F, echo=T, class.source="sccCode"}
    # for each directory in our specified condition (remember that * is a wildcard):
for dir in /Users/sofia/Documents/WoollyRun9/HQ_SRY/*/;
do
    # ${variable%something} is the part of $variable
    # before the string "something"
    # basename path/to/file is the name of the file
    # without the full path
    # $(some command) allows you to use command result as a string
    # Combining the above, we form a string based on our fasta file
    # This string can be useful to name stuff in a clean manner later
  barcode_name=$(basename ${dir%})
  echo ${barcode_name}
    # Create a directory for the results for this sequence
    # -p option avoids a failure in case the directory already exists
  mkdir -p  ~/Documents/SMAGL/Woolly_DispersalPipeline/sequencing_data/WoollyRun9/HQ_SRY/${barcode_name}
  cd ${dir}
  cat $(ls -t) > ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/HQ_SRY/${barcode_name}/HQWoollyRun9_${barcode_name}.fastq.gz;
done
```

Let's get the stats again:

```{bash HQ stats, eval=F, echo=T, class.source="terminalCode"}
for runfile in ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/HQ_SRY/*/*.fastq.gz;
do
    # Run seqkit stats on each file and append to a single document
    seqkit stats ${runfile} >> ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/HQ_SRY/HQReads_WoollyRun9_HQ1.txt
done

# Remove duplicate headers and save as new file
sed '2,${/file/d;}' ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/HQ_SRY/HQReads_WoollyRun9_HQ1.txt > ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/HQ_SRY/HQReads_WoollyRun9_HQ.txt

# Delete the temporary file
rm ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/HQ_SRY/HQReads_WoollyRun9_HQ1.txt
```

## Indexing our reference

We're using our already mapped index of the Alouatta pigra SRY sequence that we’ll use as our reference!


## Mapping to the reference

Here we map all reads to Alouatta pigra SRY, outputting them into a single sam file:
Again, I'm just doing all of them indiviudally for right now since there are only 5 and I'm on a crunch for time and don't have time to be fancy. 

```{bash mapping reference all HQ BC, eval=F, echo=T, class.source="terminalCode"}
minimap2 -ax map-ont ~/Documents/SMAGL/Woolly_DispersalPipeline/Alouatta_pigra.SRY.CortezOrtiz.fasta ~/Documents/SMAGL/Woolly_DispersalPipeline/sequencing_data/WoollyRun9/HQ_SRY/barcode12/HQWoollyRun9_barcode12.fastq.gz -a -o ~/Documents/SMAGL/Woolly_DispersalPipeline/sequencing_data/WoollyRun9/HQ_SRY/barcode12/SRY_barcode12_HQreads_vs_ApigSRY.sam -t 24


minimap2 -ax map-ont ~/Documents/SMAGL/Woolly_DispersalPipeline/Alouatta_pigra.SRY.CortezOrtiz.fasta ~/Documents/SMAGL/Woolly_DispersalPipeline/sequencing_data/WoollyRun9/HQ_SRY/barcode13/HQWoollyRun9_barcode13.fastq.gz -a -o ~/Documents/SMAGL/Woolly_DispersalPipeline/sequencing_data/WoollyRun9/HQ_SRY/barcode13/SRY_barcode13_HQreads_vs_ApigSRY.sam -t 24


minimap2 -ax map-ont ~/Documents/SMAGL/Woolly_DispersalPipeline/Alouatta_pigra.SRY.CortezOrtiz.fasta ~/Documents/SMAGL/Woolly_DispersalPipeline/sequencing_data/WoollyRun9/HQ_SRY/barcode14/HQWoollyRun9_barcode14.fastq.gz -a -o ~/Documents/SMAGL/Woolly_DispersalPipeline/sequencing_data/WoollyRun9/HQ_SRY/barcode14/SRY_barcode14_HQreads_vs_ApigSRY.sam -t 24


minimap2 -ax map-ont ~/Documents/SMAGL/Woolly_DispersalPipeline/Alouatta_pigra.SRY.CortezOrtiz.fasta ~/Documents/SMAGL/Woolly_DispersalPipeline/sequencing_data/WoollyRun9/HQ_SRY/barcode15/HQWoollyRun9_barcode15.fastq.gz -a -o ~/Documents/SMAGL/Woolly_DispersalPipeline/sequencing_data/WoollyRun9/HQ_SRY/barcode15/SRY_barcode15_HQreads_vs_ApigSRY.sam -t 24


minimap2 -ax map-ont ~/Documents/SMAGL/Woolly_DispersalPipeline/Alouatta_pigra.SRY.CortezOrtiz.fasta ~/Documents/SMAGL/Woolly_DispersalPipeline/sequencing_data/WoollyRun9/HQ_SRY/barcode16/HQWoollyRun9_barcode16.fastq.gz -a -o ~/Documents/SMAGL/Woolly_DispersalPipeline/sequencing_data/WoollyRun9/HQ_SRY/barcode16/SRY_barcode16_HQreads_vs_ApigSRY.sam -t 24
```

Sort and filter for mapped reads only and convert them to bam files:

```{bash all BC bam conversion, eval=F, echo=T, class.source="terminalCode"}
samtools sort ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/HQ_SRY/barcode12/SRY_barcode12_HQreads_vs_ApigSRY.sam -o ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/HQ_SRY/barcode12/SRY_barcode12_HQreads_vs_ApigSRY.bam 

samtools view -F 0x4 -b ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/HQ_SRY/barcode12/SRY_barcode12_HQreads_vs_ApigSRY.bam  > ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/HQ_SRY/barcode12/SRY_barcode12_HQreads_vs_ApigSRY.bam.aln.bam 

samtools index ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/HQ_SRY/barcode12/SRY_barcode12_HQreads_vs_ApigSRY.bam.aln.bam 



samtools sort ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/HQ_SRY/barcode13/SRY_barcode13_HQreads_vs_ApigSRY.sam -o ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/HQ_SRY/barcode13/SRY_barcode13_HQreads_vs_ApigSRY.bam 

samtools view -F 0x4 -b ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/HQ_SRY/barcode13/SRY_barcode13_HQreads_vs_ApigSRY.bam  > ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/HQ_SRY/barcode13/SRY_barcode13_HQreads_vs_ApigSRY.bam.aln.bam 

samtools index ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/HQ_SRY/barcode13/SRY_barcode13_HQreads_vs_ApigSRY.bam.aln.bam 



samtools sort ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/HQ_SRY/barcode14/SRY_barcode14_HQreads_vs_ApigSRY.sam -o ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/HQ_SRY/barcode14/SRY_barcode14_HQreads_vs_ApigSRY.bam 

samtools view -F 0x4 -b ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/HQ_SRY/barcode14/SRY_barcode14_HQreads_vs_ApigSRY.bam  > ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/HQ_SRY/barcode14/SRY_barcode14_HQreads_vs_ApigSRY.bam.aln.bam 

samtools index ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/HQ_SRY/barcode14/SRY_barcode14_HQreads_vs_ApigSRY.bam.aln.bam 



samtools sort ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/HQ_SRY/barcode15/SRY_barcode15_HQreads_vs_ApigSRY.sam -o ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/HQ_SRY/barcode15/SRY_barcode15_HQreads_vs_ApigSRY.bam 

samtools view -F 0x4 -b ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/HQ_SRY/barcode15/SRY_barcode15_HQreads_vs_ApigSRY.bam  > ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/HQ_SRY/barcode15/SRY_barcode15_HQreads_vs_ApigSRY.bam.aln.bam 

samtools index ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/HQ_SRY/barcode15/SRY_barcode15_HQreads_vs_ApigSRY.bam.aln.bam 



samtools sort ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/HQ_SRY/barcode16/SRY_barcode16_HQreads_vs_ApigSRY.sam -o ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/HQ_SRY/barcode16/SRY_barcode16_HQreads_vs_ApigSRY.bam 

samtools view -F 0x4 -b ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/HQ_SRY/barcode16/SRY_barcode16_HQreads_vs_ApigSRY.bam  > ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/HQ_SRY/barcode16/SRY_barcode16_HQreads_vs_ApigSRY.bam.aln.bam 

samtools index ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/HQ_SRY/barcode16/SRY_barcode16_HQreads_vs_ApigSRY.bam.aln.bam 
```


Convert alignments to FASTQ:

```{bash all BC FASTQ conversion, eval=F, echo=T, class.source="terminalCode"}
samtools bam2fq ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/HQ_SRY/barcode12/SRY_barcode12_HQreads_vs_ApigSRY.bam.aln.bam  > ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/HQ_SRY/barcode12/SRY_barcode12_HQreads_vs_ApigSRY.aln.fastq

samtools bam2fq ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/HQ_SRY/barcode13/SRY_barcode13_HQreads_vs_ApigSRY.bam.aln.bam  > ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/HQ_SRY/barcode13/SRY_barcode13_HQreads_vs_ApigSRY.aln.fastq


samtools bam2fq ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/HQ_SRY/barcode14/SRY_barcode14_HQreads_vs_ApigSRY.bam.aln.bam  > ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/HQ_SRY/barcode14/SRY_barcode14_HQreads_vs_ApigSRY.aln.fastq 


samtools bam2fq ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/HQ_SRY/barcode15/SRY_barcode15_HQreads_vs_ApigSRY.bam.aln.bam  > ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/HQ_SRY/barcode15/SRY_barcode15_HQreads_vs_ApigSRY.aln.fastq

samtools bam2fq ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/HQ_SRY/barcode16/SRY_barcode16_HQreads_vs_ApigSRY.bam.aln.bam  > ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/HQ_SRY/barcode16/SRY_barcode16_HQreads_vs_ApigSRY.aln.fastq
```


```{bash, eval = F, echo = T, class.source="terminalCode"}
seqkit stats ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/HQ_SRY/barcode12/SRY_barcode12_HQreads_vs_ApigSRY.aln.fastq

seqkit stats ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/HQ_SRY/barcode13/SRY_barcode13_HQreads_vs_ApigSRY.aln.fastq

seqkit stats ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/HQ_SRY/barcode14/SRY_barcode14_HQreads_vs_ApigSRY.aln.fastq

seqkit stats ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/HQ_SRY/barcode15/SRY_barcode15_HQreads_vs_ApigSRY.aln.fastq

seqkit stats ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/HQ_SRY/barcode16/SRY_barcode16_HQreads_vs_ApigSRY.aln.fastq

# 12: FASTQ   DNA      9,552  4,112,417      148    430.5      876
# 13: FASTQ   DNA     34,457  14,880,948      153    431.9      988
# 14: FASTQ   DNA         43   18,782      318    436.8      633
# 15: FASTQ   DNA      2,592  1,132,854      174    437.1    1,137
# 16: FASTQ   DNA        296  129,709      223    438.2      661
```


## Coverage for all BCs

 Let’s take a look at coverage for the rest of our barcdoes:

```{bash coverage all BCs, eval=F, echo=T, class.source="terminalCode"}
samtools depth ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/HQ_SRY/barcode12/SRY_barcode12_HQreads_vs_ApigSRY.bam.aln.bam > ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/HQ_SRY/barcode12/SRY_barcode12_HQreads_vs_ApigSRY.coverage


samtools depth ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/HQ_SRY/barcode13/SRY_barcode13_HQreads_vs_ApigSRY.bam.aln.bam > ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/HQ_SRY/barcode13/SRY_barcode13_HQreads_vs_ApigSRY.coverage


samtools depth ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/HQ_SRY/barcode14/SRY_barcode14_HQreads_vs_ApigSRY.bam.aln.bam > ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/HQ_SRY/barcode14/SRY_barcode14_HQreads_vs_ApigSRY.coverage


samtools depth ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/HQ_SRY/barcode15/SRY_barcode15_HQreads_vs_ApigSRY.bam.aln.bam > ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/HQ_SRY/barcode15/SRY_barcode15_HQreads_vs_ApigSRY.coverage


samtools depth ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/HQ_SRY/barcode16/SRY_barcode16_HQreads_vs_ApigSRY.bam.aln.bam > ~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/HQ_SRY/barcode16/SRY_barcode16_HQreads_vs_ApigSRY.coverage
```


Then, we import into R and do a few manipulations to graph this output so we can visualize depth of coverage across the Alouatta pigra SRY amplicon by position:

```{r coverage all BC}
HQcoverageBC12Apig=read.table("~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/HQ_SRY/barcode12/SRY_barcode12_HQreads_vs_ApigSRY.coverage", sep="\t", header=F)

HQcoverageBC13Apig=read.table("~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/HQ_SRY/barcode13/SRY_barcode13_HQreads_vs_ApigSRY.coverage", sep="\t", header=F)

HQcoverageBC14Apig=read.table("~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/HQ_SRY/barcode14/SRY_barcode14_HQreads_vs_ApigSRY.coverage", sep="\t", header=F)

HQcoverageBC15Apig=read.table("~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/HQ_SRY/barcode15/SRY_barcode15_HQreads_vs_ApigSRY.coverage", sep="\t", header=F)

HQcoverageBC16Apig=read.table("~/Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/HQ_SRY/barcode16/SRY_barcode16_HQreads_vs_ApigSRY.coverage", sep="\t", header=F)
```

Alright let's plot them all 

```{r coverage fig all BCs}
library(reshape)
library(tidyverse)


HQcoverageBC12Apig <- HQcoverageBC12Apig %>%
  rename_with(~ c("Chr", "locus", "depth"), .cols = 1:3)

ggplot(HQcoverageBC12Apig, aes(x=locus, y=depth)) + theme_bw() +
geom_point(colour="darkgreen", size=1, shape=20, alpha=1/3) +
scale_y_continuous(trans = scales::log10_trans(), breaks = scales::trans_breaks("log10", function(x) 10^x)) +
  ggtitle("HQBC12")



HQcoverageBC13Apig <- HQcoverageBC13Apig %>%
  rename_with(~ c("Chr", "locus", "depth"), .cols = 1:3)

ggplot(HQcoverageBC12Apig, aes(x=locus, y=depth)) + theme_bw() +
geom_point(colour="darkgreen", size=1, shape=20, alpha=1/3) +
scale_y_continuous(trans = scales::log10_trans(), breaks = scales::trans_breaks("log10", function(x) 10^x)) +
  ggtitle("HQBC13")


HQcoverageBC14Apig <- HQcoverageBC14Apig %>%
  rename_with(~ c("Chr", "locus", "depth"), .cols = 1:3)

ggplot(HQcoverageBC14Apig, aes(x=locus, y=depth)) + theme_bw() +
geom_point(colour="darkgreen", size=1, shape=20, alpha=1/3) +
scale_y_continuous(trans = scales::log10_trans(), breaks = scales::trans_breaks("log10", function(x) 10^x)) +
  ggtitle("HQBC14")


HQcoverageBC15Apig <- HQcoverageBC15Apig %>%
  rename_with(~ c("Chr", "locus", "depth"), .cols = 1:3)

ggplot(HQcoverageBC15Apig, aes(x=locus, y=depth)) + theme_bw() +
geom_point(colour="darkgreen", size=1, shape=20, alpha=1/3) +
scale_y_continuous(trans = scales::log10_trans(), breaks = scales::trans_breaks("log10", function(x) 10^x)) +
  ggtitle("HQBC15")


HQcoverageBC16Apig <- HQcoverageBC16Apig %>%
  rename_with(~ c("Chr", "locus", "depth"), .cols = 1:3)

ggplot(HQcoverageBC16Apig, aes(x=locus, y=depth)) + theme_bw() +
geom_point(colour="darkgreen", size=1, shape=20, alpha=1/3) +
scale_y_continuous(trans = scales::log10_trans(), breaks = scales::trans_breaks("log10", function(x) 10^x)) +
  ggtitle("HQBC16")
```

# making consensus sequences for all Barcodes
again, I ran these chunks of code on the SMAGL laptop because I don't have NGSpeciesID downloaded on my personal laptop.

```{bash all BC consensus, eval=F, echo=T, class.source="terminalCode}
conda activate NGSpeciesID

NGSpeciesID --ont --m 300 --s 150 --consensus --medaka --primer_file  /Users/smaglschmittlab/Documents/SMAGL/Woolly_DispersalPipeline/SRYprimers.txt \
--fastq /Users/smaglschmittlab//Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/HQ_SRY/barcode12/SRY_barcode12_HQreads_vs_ApigSRY.aln.fastq \
--outfolder /Users/smaglschmittlab/Documents/SMAGL/Woolly_DispersalPipeline/sequencing_data/WoollyRun9/HQ_SRY/barcode12/barcode12HQ_Apig_300e150_consensus


NGSpeciesID --ont --m 300 --s 150 --consensus --medaka --primer_file  /Users/smaglschmittlab/Documents/SMAGL/Woolly_DispersalPipeline/SRYprimers.txt \
--fastq /Users/smaglschmittlab//Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/HQ_SRY/barcode13/SRY_barcode13_HQreads_vs_ApigSRY.aln.fastq \
--outfolder /Users/smaglschmittlab/Documents/SMAGL/Woolly_DispersalPipeline/sequencing_data/WoollyRun9/HQ_SRY/barcode13/barcode13HQ_Apig_300e150_consensus


NGSpeciesID --ont --m 300 --s 150 --consensus --medaka --primer_file  /Users/smaglschmittlab/Documents/SMAGL/Woolly_DispersalPipeline/SRYprimers.txt \
--fastq /Users/smaglschmittlab//Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/HQ_SRY/barcode14/SRY_barcode14_HQreads_vs_ApigSRY.aln.fastq \
--outfolder /Users/smaglschmittlab/Documents/SMAGL/Woolly_DispersalPipeline/sequencing_data/WoollyRun9/HQ_SRY/barcode14/barcode14HQ_Apig_300e150_consensus


NGSpeciesID --ont --m 300 --s 150 --consensus --medaka --primer_file  /Users/smaglschmittlab/Documents/SMAGL/Woolly_DispersalPipeline/SRYprimers.txt \
--fastq /Users/smaglschmittlab//Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/HQ_SRY/barcode15/SRY_barcode15_HQreads_vs_ApigSRY.aln.fastq \
--outfolder /Users/smaglschmittlab/Documents/SMAGL/Woolly_DispersalPipeline/sequencing_data/WoollyRun9/HQ_SRY/barcode15/barcode15HQ_Apig_300e150_consensus


NGSpeciesID --ont --m 300 --s 150 --consensus --medaka --primer_file  /Users/smaglschmittlab/Documents/SMAGL/Woolly_DispersalPipeline/SRYprimers.txt \
--fastq /Users/smaglschmittlab//Documents/SMAGL/woolly_DispersalPipeline/sequencing_data/WoollyRun9/HQ_SRY/barcode16/SRY_barcode16_HQreads_vs_ApigSRY.aln.fastq \
--outfolder /Users/smaglschmittlab/Documents/SMAGL/Woolly_DispersalPipeline/sequencing_data/WoollyRun9/HQ_SRY/barcode16/barcode16HQ_Apig_300e150_consensus


#closing out conda
conda deactivate
```


## Alignment 

Let's align our HQ sequences 
```{r}
#This loop creates a list of all the medaka-polished consensus sequences in our barcode folders:
barcodes<-c("barcode12","barcode13","barcode14","barcode15","barcode16")

library(Biostrings)
HQ.multibarcode.300_150<-c()
for (barcode in barcodes) {
  a<-paste0("~/Documents/SMAGL/Woolly_DispersalPipeline/sequencing_data/WoollyRun9/HQ_SRY/",barcode,
            "/",barcode,"HQ_Apig_300e150_consensus/medaka_cl_id_*/consensus.fasta")

  
  assign(paste0(barcode,"_150con"),Biostrings::readDNAStringSet(Sys.glob(a)))
  
  HQ.multibarcode.300_150<-c(HQ.multibarcode.300_150,get(paste0(barcode,"_150con")))  
}

#And this concatenates (unlists) all the FASTA files into one file:
names(HQ.multibarcode.300_150) <- NULL
HQ.multibarcode.300_150<-do.call(c, HQ.multibarcode.300_150)
names(HQ.multibarcode.300_150)<-c("barcode12","barcode13","barcode14","barcode15","barcode16")

HQ.multibarcode.300_150
```

We can also see which are identical to each other:
```{r}
duplicated(HQ.multibarcode.300_150)
```
oo there are duplicates now. tea..

### Aligning All Seqs

```{r}
#And now we can make an alignment:
aln.HQ.barcode300_150<-rMSA::mafft(HQ.multibarcode.300_150)
```
### Phylogeny of All HQ Seqs

Let's make a new tree with the HQ Seqs
```{r}
library(phangorn)
HQ.barcode300_150.phyDat<-as.phyDat(aln.HQ.barcode300_150)
dm  <- dist.ml(HQ.barcode300_150.phyDat)
treeUPGMA  <- upgma(dm)
treeNJ  <- NJ(dm)
fun <- function(x) { upgma(dist.ml(x)) }
bs_upgma <- bootstrap.phyDat(HQ.barcode300_150.phyDat, fun)
plotBS(treeUPGMA, bs_upgma, main="UPGMA of HQ SRY 300-150")
```
waaaiiittt this is teaaa. ok interesting. this seems more informative than our last tree. 
here are were the samples are from again, just for reference: 

12: Uchiza
13: Uchiza
14: Corosha
15: Mariposa
16: Uchiza

### Prettifying our Alignment

We can take a closer look at the alignment here. We'll use the [{ggmsa}](http://yulab-smu.top/ggmsa/) package:
```{r,warning=F}
library(ggmsa)

#png(filename="~/Documents/SMAGL/Woolly_DispersalPipeline/sequencing_data/WoollyRun9/HQ_SRY/HQ_SRY_con300_150_aln.trim.png",width = 25, height = 5, units = "in",res=600)

ggmsa::ggmsa(aln.HQ.barcode300_150, seq_name = T, color = "Chemistry_NT") + geom_seqlogo(color = "Chemistry_NT") + geom_msaBar()

dev.off()
```


### Reverse Complementing Seqs
alright barcodes 12 and 14 kinda look funky so let's try reversing them. I use the same motif that Chris does in his starter pipeline to automate the detection of likely reverse transcripts.


```{r}
#Let's try automating the detection of likely reverse transcripts. We'll do this by looking for a common DNA motif across all transcripts of one orientation that doesn't contain any indels. Note that you should visually check any new alignment to estimate what the proper motif of the forward strand should be, as there may be indels in your previous motif in new samples.

#This code automatically detects likely reverse complements by noting which sequences are missing the motif 'TTTCCAACTCTGTCAT', which is shared across most sequences. It then isolates them in its multisequence object and reverse complements them all at once:
HQbarcodes150rc<-reverseComplement(HQ.multibarcode.300_150[which(vcountPattern('TTTCCAACTCTGTCAT', HQ.multibarcode.300_150) < 1)])

names(HQbarcodes150rc)<-paste0(names(HQbarcodes150rc),".rc") #rename seqs with ".rc" at the end

#Let's remove the ones we've reverse complemented from our original object:
HQ.multibarcode.300_150.fw<-HQ.multibarcode.300_150[which(vcountPattern('TTTCCAACTCTGTCAT', HQ.multibarcode.300_150) >= 1)]

#And this concatenates our two objects, the reversed and the original minus reversed:
HQ.multibarcode.300_150.rc<-c(HQ.multibarcode.300_150.fw,HQbarcodes150rc)

#And here we can reorder the sequences so that same-barcodes align:
HQ.multibarcode.300_150.rc<-HQ.multibarcode.300_150.rc[order(names(HQ.multibarcode.300_150.rc))]
```


Ok let's realign to see how it's looking. 

### Re-Aligning with RC
```{r}
HQ.multibarcode.300_150.rc.aln <- rMSA::mafft(HQ.multibarcode.300_150.rc)
HQ.multibarcode.300_150.rc.aln
```

Ok this looks better. 12 and 14 were on the reverse strand again (as expected) 

```{r}
HQ.multibarcode.300_150.rc.plot <- ggmsa::ggmsa(HQ.multibarcode.300_150.rc.aln, seq_name = T, color = "Chemistry_NT") + geom_seqlogo(color = "Chemistry_NT") + geom_msaBar()
HQ.multibarcode.300_150.rc.plot

#Below is code to save the alignment image:
png(filename="~/Documents/SMAGL/Woolly_DispersalPipeline/sequencing_data/WoollyRun9/HQ_SRY/HQ_SRY_con300_150_aln.trim.png",width = 30, height = 7, units = "in",res=600)
HQ.multibarcode.300_150.rc.plot
dev.off()
```

Again, don't think this needs trimming. let's keep going. 

### New Phylogeny
```{r}
library(phangorn)

HQ_barcode300_150.phyDat<-as.phyDat(HQ.multibarcode.300_150.rc.aln)
dm  <- dist.ml(HQ_barcode300_150.phyDat)
treeUPGMA  <- upgma(dm)
treeNJ  <- NJ(dm)
fun <- function(x) { upgma(dist.ml(x)) }
bs_upgma <- bootstrap.phyDat(HQ_barcode300_150.phyDat, fun)
plotBS(treeUPGMA, bs_upgma, main="UPGMA of HQ SRY 300-150")
```
uh oh.

## Haplotype Network

```{r}
library(tidyverse)
library(Biostrings)

HQlagoSRY12<-readDNAStringSet("/Users/sofia/Documents/SMAGL/Woolly_DispersalPipeline/sequencing_data/WoollyRun9/HQ_SRY/barcode12/barcode12HQ_Apig_300e150_consensus/medaka_cl_id_9/consensus.fasta")

HQlagoSRY13<-readDNAStringSet("/Users/sofia/Documents/SMAGL/Woolly_DispersalPipeline/sequencing_data/WoollyRun9/HQ_SRY/barcode13/barcode13HQ_Apig_300e150_consensus/medaka_cl_id_23/consensus.fasta")

HQlagoSRY14<-readDNAStringSet("/Users/sofia/Documents/SMAGL/Woolly_DispersalPipeline/sequencing_data/WoollyRun9/HQ_SRY/barcode14/barcode14HQ_Apig_300e150_consensus/medaka_cl_id_2/consensus.fasta")

HQlagoSRY15<-readDNAStringSet("/Users/sofia/Documents/SMAGL/Woolly_DispersalPipeline/sequencing_data/WoollyRun9/HQ_SRY/barcode15/barcode15HQ_Apig_300e150_consensus/medaka_cl_id_16/consensus.fasta")

HQlagoSRY16<-readDNAStringSet("/Users/sofia/Documents/SMAGL/Woolly_DispersalPipeline/sequencing_data/WoollyRun9/HQ_SRY/barcode16/barcode16HQ_Apig_300e150_consensus/medaka_cl_id_3/consensus.fasta")


library(Biostrings)

# Combine all consensus sequences into a single DNAStringSet

names(HQlagoSRY12) <- "barcode12"
names(HQlagoSRY13) <- "barcode13"
names(HQlagoSRY14) <- "barcode14"
names(HQlagoSRY15) <- "barcode15"
names(HQlagoSRY16) <- "barcode16"


consensus_combined <- c(HQlagoSRY12, HQlagoSRY13, HQlagoSRY14, HQlagoSRY15, HQlagoSRY16)


# Write to a new FASTA file
writeXStringSet(consensus_combined, filepath = "/Users/sofia/Documents/SMAGL/Woolly_DispersalPipeline/sequencing_data/HQ_SRY_combined_consensus.fasta")
```


```{r}
library(DECIPHER)

HQlagoSRY<-readDNAStringSet("~/Documents/SMAGL/Woolly_DispersalPipeline/sequencing_data/HQ_SRY_combined_consensus.fasta")

HQlagoSRY <- OrientNucleotides(HQlagoSRY)

#Run the alignment
HQlagoSRY.aln.pre <- AlignSeqs(c(HQlagoSRY))
```

```{r}
length(HQlagoSRY.aln.pre)
HQlagoSRY.aln.pre
```

```{r}
SRY.popinfo <-read.csv("~/Documents/SMAGL/woolly_DispersalPipeline/SRY_popinfo.csv", sep = ",", header=TRUE, comment.char = "#")
rownames(SRY.popinfo)<-SRY.popinfo$barcode
SRY.popinfo
```
aligning

```{r}
HQlagoSRY.aln<-HQlagoSRY.aln.pre[SRY.popinfo$barcode]
length(HQlagoSRY.aln)

HQlagoSRY.aln
```

running into some errors with the chunk of code to get our haplotypes. so i'm trimming the very front and very end of our sequences because it seems to be cranky with NA values (even though this wasn't an issue the first time)


```{r}
BrowseSeqs(HQlagoSRY.aln, highlight=0)


HQlagoSRY.aln.trim<-subseq(HQlagoSRY.aln,start=2,end=274)
BrowseSeqs(HQlagoSRY.aln.trim, highlight=0)

HQlagoSRY.aln.trim
```


Now we’ll load the package we’ll use to construct our haplotype network, called {geneHapR}:

```{r}
library(geneHapR)

HQ.SRYhapResult <- seqs2hap(HQlagoSRY.aln.trim,
                      Ref = names(HQlagoSRY.aln.trim)[1],
                      hapPrefix = "H",
                      hetero_remove = TRUE,
                      na_drop = TRUE,
                      maxGapsPerSeq = 0.25)

sites(HQ.SRYhapResult)
```


Summary of SRY haplotypes

```{r}
HQ.SRYhapSummary <- hap_summary(HQ.SRYhapResult)
print(HQ.SRYhapSummary)
```
uhhh it's saying they all have their own haplotype? I think this will change once we get the HQ data. 

```{r}
plotHapTable(HQ.SRYhapSummary,
             title = "Lagothrix flavicauda SRY Haplotypes")
```

```{r}
HQ.SRYhapNet <- get_hapNet(HQ.SRYhapSummary,
                     AccINFO = SRY.popinfo,
                     groupName = "collection_site")
```


```{r}

plot(HQ.SRYhapNet)


plotHapNet(HQ.SRYhapNet,
           size = "freq",                   # circle size
           scale = "log10",                 # scale circle with 'log10(size + 1)'
           cex = 1,                         # size of hap symbol
           col.link = 1,                    # link colors
           link.width = 1,                  # link widths
           show.mutation = 2,               # mutation types one of c(0,1,2,3)
           #hapGroup = 61,                   # draw pie charts for each haplotype
           legend = c(6,6),                # legend position
           show_color_legend = TRUE,
           show_size_legend = TRUE,
           labels = FALSE,
           pie.lim = c(2, 5),
           backGround=c("red3","orange","skyblue"))
```




## mtDNA time

ok. let's import Mel's platy3 consensus sequence data 

```{r}
library(tidyverse)
library(Biostrings)
library(DECIPHER)


lagoplaty3 <-readDNAStringSet("~/Documents/SMAGL/Woolly_DispersalPipeline/sequencing_data/lagflav_platy3_filt_mar27.fasta")

lagoplaty3 <-OrientNucleotides(lagoplaty3)

#Run the alignment
lagoplaty3.aln.pre <- AlignSeqs(c(lagoplaty3))

```

seeing how many sequences we have

```{r}
length(lagoplaty3.aln.pre)
lagoplaty3.aln.pre
```
eek I'm definitely going to have to trim.. let's take a look

```{r}
BrowseSeqs(lagoplaty3.aln.pre, highlight=0)
```


```{r}
lagoplaty3.aln.trim <- subseq(lagoplaty3.aln.pre,start=241,end=10049) #trimming

lagoplaty3.aln.trim <- lagoplaty3.aln.trim[!names(lagoplaty3.aln.trim) %in% c("LF20_platy3", "LF79_platy3", "LF129_platy3")] # Remove sequences by name that have large chunks missing

#BrowseSeqs(lagoplaty3.aln.trim, highlight=0)

lagoplaty3.aln.trim
```


```{r}
platy3popinfo <- read.csv("~/Documents/SMAGL/Woolly_DispersalPipeline/assembly_info_platy3_platy4_allruns.csv", sep = ",", header = TRUE)
platy3popinfo
```
ok this includes everyone.. so let's just keep the ones that we have sequences for and remove the duplicate LF110 that there is. I kept contig_2 because the coverage was better (24 vs 17)

```{r}
# Get the sample names from your DNAStringSet object
sample <- names(lagoplaty3.aln.trim)

# Filter the CSV rows to only keep those with matching sample names
platy3popinfo.filt <- platy3popinfo[platy3popinfo$Sample %in% sample & !(platy3popinfo$sample_code == "LF110" & platy3popinfo$X.seq_name == "contig_1"), ]

platy3popinfo.filt
```


```{r}
rownames(platy3popinfo.filt) <- platy3popinfo.filt$Sample
platy3popinfo.filt
```



```{r}
lagoplaty3.aln <- lagoplaty3.aln.trim[platy3popinfo.filt$Sample]
length(lagoplaty3.aln)

lagoplaty3.aln

#BrowseSeqs(lagoplaty3.aln, highlight=0) checking it out to see if there's anything weird

```
ok so i think there is some sort of issue with this because my DNAStringSet looks correct... not really sure what to do. 

```{r}
library(geneHapR)
p3hapResult <- seqs2hap(lagoplaty3.aln,
                      Ref = names(lagoplaty3.aln)[1],
                      hapPrefix = "H",
                      hetero_remove = TRUE,
                      na_drop = TRUE,
                      maxGapsPerSeq = 0.25)
```


# see how many informing sites for our haplotypes 

```{r}
sites(p3hapResult) #segregating sites

# [1] 2689 hm..
```


summary of haplotypes

```{r}
p3hapSummary <- hap_summary(p3hapResult)
print(p3hapSummary)
```
ok so i can't really tell if there's a problem here. There's no accession or frequency columns in my summary like there was for SRY? sigh. 

Plotting a haplotype table just cause
```{r}
# Remove columns where all values are NA
plotHapTable(p3hapSummary,
             title = "Lagothrix lagothricha Platy3 Haplotypes")
```


Now let’s calculate our haplotype network, importing our filered platy3popinfo so that we can group our haplotypes by site:

```{r}
p3hapNet <- get_hapNet(p3hapSummary,
                     AccINFO = platy3popinfo.filt,
                     groupName = "site")


```


plotting a figure 

```{r}
plotHapNet(p3hapNet,
           size = "freq",                   # circle size
           scale = "log10",                 # scale circle with 'log10(size + 1)'
           cex = 1,                       # size of hap symbol
           col.link = 1,                    # link colors
           link.width = 1,                  # link widths
           show.mutation = 2,               # mutation types one of c(0,1,2,3)
           #hapGroup = 61,                   # draw pie charts for each haplotype
           legend = c(-15, 60),            # legend position
           show_color_legend = TRUE,
           show_size_legend = TRUE,
           labels = FALSE,
           pie.lim = c(2, 5),
           backGround=c("yellow","skyblue","purple4","limegreen","darkblue","purple","orange","red3"))  
```





title: "Woolly Dispersal Pipeline"
author: "Sofia M. Weaver"
date: "`r Sys.Date()`"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## R Markdown

This is an R Markdown document. Markdown is a simple formatting syntax for authoring HTML, PDF, and MS Word documents. For more details on using R Markdown see <http://rmarkdown.rstudio.com>.

When you click the **Knit** button a document will be generated that includes both content as well as the output of any embedded R code chunks within the document. You can embed an R code chunk like this:

```{r cars}
summary(cars)
```

## Including Plots

You can also embed plots, for example:

```{r pressure, echo=FALSE}
plot(pressure)
```

Note that the `echo = FALSE` parameter was added to the code chunk to prevent printing of the R code that generated the plot.
